extensions/narayana-jta/runtime/src/main/java/io/quarkus/narayana/jta/runtime/NotifyingUserTransaction.java

```diff
@@ -0,0 +1,68 @@
+package io.quarkus.narayana.jta.runtime;
+
+import javax.transaction.HeuristicMixedException;
+import javax.transaction.HeuristicRollbackException;
+import javax.transaction.NotSupportedException;
+import javax.transaction.RollbackException;
+import javax.transaction.SystemException;
+import javax.transaction.UserTransaction;
+
+import org.jboss.logging.Logger;
+
+public class NotifyingUserTransaction extends TransactionScopedNotifier implements UserTransaction {
```

Comment:
Note that the `com.arjuna.ats.internal.jta.transaction.arjunacore.UserTransactionImple` that was used directly before and is used under the hood in this PR implements `Serializable`. So in theory this is a breaking change in the sense that a user might expect the contextual instance to be `Serializable`.

I tend not to implement `Serializable` for several reasons:
- Quarkus does not support CDI passivation and passivation scopes
- the producer bean for `UserTransaction` is `@ApplicationScoped` which means that a non-serializable client proxy is injected, i.e. a user would need to unwrap the bean instance first and then serialize (I doubt anyone does it)
- `javax.transaction.UserTransaction` is not serializable and the contract of the producer bean is clear - the set of bean types contains the return type and all interfaces it extends 

WDYT? @mmusgrov @manovotn

Reply:
I agree @mkouba - passivating a transaction does not make sense. Thanks for sorting out this issue.

Reply:
Agreed, but shouldn't we be consistent in how we do this for `NotifyingTransactionManager` and `NotifyingUserTransaction`? Currently, one is serializable while the other isn't.

Reply:
I don't think it's a problem because the serializability is not a part of the API contract.